core/src/test/java/org/elasticsearch/aliases/IndexAliasesIT.java

```diff
@@ -427,6 +428,23 @@ public void testDeleteAliases() throws Exception {
 
         AliasesExistResponse response = admin().indices().prepareAliasesExist(aliases).get();
         assertThat(response.exists(), equalTo(false));
+
+        logger.info("--> creating index [foo_foo] and [bar_bar]");
+        assertAcked(prepareCreate("foo_foo"));
+        assertAcked(prepareCreate("bar_bar"));
+        ensureGreen();
+
+        logger.info("--> adding [foo] alias to [foo_foo] and [bar_bar]");
+        assertAcked(admin().indices().prepareAliases().addAlias("foo_foo", "foo"));
+        assertAcked(admin().indices().prepareAliases().addAlias("bar_bar", "foo"));
+
+        assertAcked(admin().indices().prepareAliases().addAliasAction(AliasActions.remove().index("foo*").alias("foo")).execute().get());
```

Comment:
Unfortunately the `GetAliasesRequest` is not yet fixed ( will be done in a separate PR ) so I don't think that it would bring anything adding it here at this stage as it will resolve to both indices ...

But I do agree that it would be nice to add this check once the `GetAliasesrequest` is changed ;)